windows/src/engine/keyman32/keyboardoptions.cpp

```diff
@@ -30,6 +32,10 @@ void LoadKeyboardOptions(LPINTKEYBOARDINFO kp)
 
 void LoadSharedKeyboardOptions(LPINTKEYBOARDINFO kp)
 {
+  if (Globals::get_CoreIntegration())
+  {
+    return; // should never be called if using core processor
```

Comment:
Add a debug assertion.

Reply:
I added debug messages for all these 
``SendDebugMessageFormat(0, sdmAIDefault, 0, "LoadSharedKeyboardOptions: Error called in core integration mode");``
as you said not to use asset here
"oh, don't put assertions into there -- they'll get blocked anyway and Windows will silently unload the DLL"

Reply:
Good point on `assert`, we should do them with logging. Perhaps define our own assert func that does that? e.g.

```cpp
#define assertOrReturnFalse(...) { if(...) { SendDebug...; return FALSE; } }
#define assertOrReturn(...) { if(...) { SendDebug...; return; } }
```